Change SpriteFont.xnb to support ar, de, kr, ru, tr, vi, zh,...

ApocBruh opened the pull request:
Co-authored by: MantisMO <98868186+MantisMO@users.noreply.github.com>

ApocBruh committed:
Change SpriteFont.xnb to support ar, de, kr, ru, tr, vi, zh,...

Co-authored by: MantisMO <98868186+MantisMO@users.noreply.github.com>

SadPencil commented:
> XNA build cannot work with updated fonts, it crashes on the start.

https://github.com/CnCNet/xna-cncnet-client/pull/830

pzhlkj6612 commented:
Off-topic:

Quick tip for @ApocBruh: To add co-authors, use `Co-authored-by: name <email>` in commit message, then you'll see their avatar appears. Please note the two hyphens. Read [Creating co-authored commits on the command line § Creating a commit with multiple authors - GitHub Docs](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line).

ApocBruh committed:
Change SpriteFont.xnb to support ar, de, jp, ru, tr, vi, zh,...

Due to some technical errors, clientxna.exe cannot load the SpriteFont file. I have rebuilt the SpriteFont file with a 2048x2048 texture (XNA compatible). This rebuild will remove Korean characters and quite a few Chinese characters (Not remove the characters that MantisMO added)

Co-authored-by: MantisMO <98868186+MantisMO@users.noreply.github.com>

ApocBruh committed:
Change SpriteFont.xnb to support ar, de, jp, ru, tr, vi, zh,...

Due to some technical errors, clientxna.exe cannot load the SpriteFont file. I have rebuilt the SpriteFont file with a 2048x2048 texture (XNA compatible). This rebuild will remove Korean characters and quite a few Chinese characters (Not remove the characters that MantisMO added)

Co-authored by: MantisMO <98868186+MantisMO@users.noreply.github.com>

ApocBruh commented:
> Off-topic:
> 
> Quick tip for @ApocBruh: To add co-authors, use `Co-authored-by: name <email>` in commit message, then you'll see their avatar appears. Please note the two hyphens. Read [Creating co-authored commits on the command line § Creating a commit with multiple authors - GitHub Docs](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line).
I tried that but it didn't show up, maybe I'm wrong somewhere.
<img width="476" height="71" alt="image" src="https://github.com/user-attachments/assets/ecd5ffd2-78e0-4693-a531-f802391e7237" />

ApocBruh committed:
Merge remote-tracking branch 'origin/spritefont' into spritefont

ApocBruh commented:
> Off-topic:
> 
> Quick tip for @ApocBruh: To add co-authors, use `Co-authored-by: name <email>` in commit message, then you'll see their avatar appears. Please note the two hyphens. Read [Creating co-authored commits on the command line § Creating a commit with multiple authors - GitHub Docs](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line).

It works, thanks!

ApocBruh committed:
Change SpriteFont.xnb to support ar, de, jp, ru, tr, vi, zh,...

Update some stuff.
Co-authored-by: MantisMO <98868186+MantisMO@users.noreply.github.com>

SadPencil commented:
Can you leave some screenshots (not from your machine) proving that the XNA variant of the client works and does not crash anymore?

ApocBruh commented:
> Can you leave some screenshots (not from your machine) proving that the XNA variant of the client works and does not crash anymore?

I will ask HuyDepTrai to help with that.

vohonghuy123 commented:
> Can you leave some screenshots (not from your machine) proving that the XNA variant of the client works and does not crash anymore?
Here with our spritefont for Vietnamese language pack work fine on clientdx.exe
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/022749b3-1493-4395-9055-c3860f4bff9e" />

vohonghuy123 commented:
> Can you leave some screenshots (not from your machine) proving that the XNA variant of the client works and does not crash anymore?

clientogl.exe also working good 
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/f5153d86-ad56-4350-829e-857e94ad4ea2" />

SadPencil commented:
> > Can you leave some screenshots (not from your machine) proving that the XNA variant of the client works and does not crash anymore?
> 
> Here with our spritefont for Vietnamese language pack work fine on clientdx.exe
> 
> <img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/022749b3-1493-4395-9055-c3860f4bff9e" />
> 
> 

the XNA variant

vohonghuy123 commented:
> Can you leave some screenshots (not from your machine) proving that the XNA variant of the client works and does not crash anymore?

clientxna.exe also working like a dream 
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/e85c7e32-3d65-4fa3-bc30-ea8ded20e4dd" />

SadPencil commented:
> > Can you leave some screenshots (not from your machine) proving that the XNA variant of the client works and does not crash anymore?
> 
> 
> 
> clientxna.exe also working like a dream 
> 
> <img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/e85c7e32-3d65-4fa3-bc30-ea8ded20e4dd" />
> 
> 

I need to confirm: have you replaced the font files, or not?

vohonghuy123 commented:
> > > Can you leave some screenshots (not from your machine) proving that the XNA variant of the client works and does not crash anymore?
> > 
> > 
> > clientxna.exe also working like a dream
> > <img alt="image" width="1920" height="1080" src="https://private-user-images.githubusercontent.com/229754758/491378325-e85c7e32-3d65-4fa3-bc30-ea8ded20e4dd.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgyNTMxNzIsIm5iZiI6MTc1ODI1Mjg3MiwicGF0aCI6Ii8yMjk3NTQ3NTgvNDkxMzc4MzI1LWU4NWM3ZTMyLTNkNjUtNGZhMy1iYzMwLWVhOGRlZDIwZTRkZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxOVQwMzM0MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2M3Njc0MTRkNDlkODdjZDVlODMzNDE2MWU5ZmVhZGFlMWJhNDMwZmRlZDZiZDI5Nzg2YjE3YjZmNmEzNjQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.gmltCjtaIsW9TIF_SkWWy5qLBKKZw1COJVyqzGFmXL8">
> 
> I need to confirm: have you replaced the font files, or not?

I just replaced 2 spritefont files about ten minutes ago. Sent by ApocBruh himself. 
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/645dcb53-d72f-4269-b666-24aff9a9c29f" />

SadPencil commented:
<img width="3204" height="1860" alt="图片" src="https://github.com/user-attachments/assets/557b6cb5-a954-44bf-b797-5e718c0f873a" />
<img width="3204" height="1860" alt="图片" src="https://github.com/user-attachments/assets/fe7711b5-ad13-4fe3-8073-240bdf58ed30" />
<img width="3204" height="1860" alt="图片" src="https://github.com/user-attachments/assets/586f0626-3e17-4199-bc65-b0a3b039b1c3" />
<img width="3204" height="1860" alt="图片" src="https://github.com/user-attachments/assets/0dd4d9fc-bbc7-438e-a95f-ef7004efc2f5" />

Not a perfect one as some commonly used zh-Hant characters are missing. I can tolerate this issue. However:

More importantly, one character *in a pair* is missing while the other one does not, indicating a strong inconsistency

<img width="1677" height="135" alt="图片" src="https://github.com/user-attachments/assets/039e34cf-ee5e-47d7-b74a-7f86d8cd7a39" />

```
在中国大陆地区，横排文稿一般使用 “ ” 和 ‘ ’，竖（直）排文稿改用『』和「」（直角引号）；在港澳台地区，则普遍使用『』和「」。
```

Character `‘` and `’` are in pairs. Same for `「` and `」`

I can accept some characters such as Korean ones and zh-Hant ones being removed due to the 2048x2048 limitation. However, for these symbols (not limited to the ones above), please retain them.
